### PR TITLE
GetAnyAgent returns random agent

### PR DESCRIPTION
**Pull request by dankraw**
...and some minor changes.

### CODE REVIEW

File: consul/agents.go [hunk 1]
```diff
@@ -31,38 +33,48 @@ func (a *ConcurrentAgents) GetAnyAgent() (*consulapi.Client, error) {
...
 	return nil, fmt.Errorf("No agent available")
 }
 
-func (a *ConcurrentAgents) GetAgent(agentAddress string) (*consulapi.Client, error) {
+func (a *ConcurrentAgents) getRandomAgentHost() string {
+	hosts := []string{}
+	for host, _ := range a.agents {
+		hosts = append(hosts, host)
+	}
+	rand.Seed(time.Now().UTC().UnixNano())
```
Comment (janisz) [inline consul/agents.go]: Seed is set every time this function is called. Better set it once in other case we could always end up with same host

Comment (dankraw) [inline consul/agents.go]: I've removed the seeding line as it should work good enough by default.

**Comment by janisz** [comment]
You can add more tests for to check error handling

File: consul/consul_test.go [hunk 2]
```diff
@@ -55,23 +55,81 @@ func TestRegisterServices(t *testing.T) {
...
+
+	// given
+	server.AddService("serviceA", "passing", []string{"marathon"})
+
+	// when
+	server.Stop()
+	err := consul.Deregister("serviceA", server.Config.NodeName)
+
+	// then
+	assert.NotNil(t, err)
```
Comment (janisz) [inline consul/consul_test.go]: `assert.Error` will be better here

**Pull request merged**